### PR TITLE
Tweak widths of plots and make easier to maintain

### PR DESCRIPTION
**Pull request by Alanocallaghan**
The idea of this PR is to refactor waterfall_align and related functions to allow the elements of the waterfall plots to be better and more tightly aligned, and to be explicit in variable names, as well as reducing some code duplication.

I can add some comments explaining the various edits in the coming days if needed

### CODE REVIEW

**Comment by zlskidmore** [comment]
Thanks @Alanocallaghan I'll give it a look this weekend!

File: R/waterfall_align.R [hunk 1]
```diff
@@ -31,81 +31,127 @@ waterfall_align <- function(p2, p1, p3, p4, p5,
...
+    genes_grob <- suppressWarnings(ggplot2::ggplotGrob(genes))
+
+    heatmap_grob <- ggplot2::ggplotGrob(heatmap)
+    
+    ## Strip out legends and plot separately
+    ## https://github.com/baptiste/gridextra/wiki/arranging-ggplot#legends
+    ind_legend <- grep("guide", heatmap_grob$layout$name)
+    heatmap_legend <- heatmap_grob[["grobs"]][[ind_legend]]
+    heatmap_width <- sum(heatmap_legend$width)
+    heatmap_grob <- ggplot2::ggplotGrob(heatmap + theme(legend.position="none"))
```
Comment (Alanocallaghan) [inline R/waterfall_align.R]: Moving the legends to a separate column of grobs means that they will never overlap the plots or spill over the side, and likewise will only take up as much room as they need

File: R/waterfall_buildMain.R [hunk 2]
```diff
@@ -84,62 +82,36 @@ waterfall_buildMain <- function(data_frame, grid=TRUE, label_x=FALSE,
 
     # Theme, Boolean, if specified to plot x labels, define theme such that
     # labels are plotted
-    if(label_x == TRUE & plot_x_title == TRUE)
-    {
-        theme <-  theme(axis.ticks=element_blank(),
+    default_theme <- theme(axis.ticks=element_blank(),
```
Comment (Alanocallaghan) [inline R/waterfall_buildMain.R]: This function is easier to maintain if duplicated theme elements are separated out (easier to change defaults, etc). Similarly, you could probably move some duplicated elements of the if/else below such that each option is only specified once

File: R/waterfall_build_proportions.R [hunk 3]
```diff
@@ -39,9 +39,9 @@ waterfall_build_proportions <- function(data_frame, plot_palette,
         theme(
             axis.ticks.x = element_blank(),
             axis.text.x =  element_blank(),
-            axis.title.x =if (x_label) element_text() else element_blank(),
+            axis.title.x = if (x_label) element_text() else element_blank(),
```
Comment (Alanocallaghan) [inline R/waterfall_build_proportions.R]: Moving the gene labels from the main heatmap/mutational landscape to the gene plot means that the axis titles for the mutational burden and clinical/proportion plots will be closer to their axes

**Comment by Alanocallaghan** [comment]
Cheers. 

I also tried to use `unit.pmin` instead of `unit.pmax` for setting the plot widths, and the result is quite nice, but there's a strong tendency for the gene plot axis title to overlap the main plot. Also, the y axis labels for the top and bottom plots tend to go very close to the y-axis. It might be possible to select just the elements to resize using a similar trick to the one I used for the legends in `waterfall_align.R` (grepping the grob layout).

**Comment by Alanocallaghan** [comment]
Also, I added a .travis.yml, you could make an account with the griffithlab github (it's free and requires no setup) and it should tell you whether the package currently passes R CMD check with bioc-release and bioc-devel. Quite handy for making sure edits aren't going to break anything

**Comment by malachig** [comment]
Thanks for the contribution @Alanocallaghan !

**Comment by Alanocallaghan** [comment]
Cheers. Incidentally you could use a similar trick to ensure bottom legends don't spill out the bottom of the plot

**Pull request merged**